### ISSUE TITLE
Inconsistent formatting of nested operators

### ISSUE THREAD

**Issue by michalmuskala** [elixir-lang/elixir#10830]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 
```
Erlang/OTP 24 [DEVELOPMENT] [erts-11.1.7] [source-af632ec337] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]

Elixir 1.11.4
```
* Operating system: macOS

### Current behavior

Formatting of the right side of `=` operator sometimes looks strange in chained operators:

```elixir
# expected, each operator indents by 1 level
a =
  b = [
    c,
    d
  ]

# also expected, each operator indents by 1 level
a ::
  b = [
    c,
    d
  ]

# unexpected! looks like the = did not indent
a ++
  b = [
  c,
  d
]
```

### Expected behavior

The behaviour should be consistent between operators

**Comment by josevalim** [comment]
This is correct I believe. The difference here is precedence. For the `++`, we are indenting the left side. Then we don't indent the right side due to our feature where we don't indent if the expression is the opening of a list, `fn`, `do`, etc.

**Comment by michalmuskala** [comment]
Yes, that feature in general makes sense, but in other places it considers the indentation of where it "starts" and does not de-dent outside of the start position:

```elixir
foo(
  a ++
    b = [
    c,
    d
  ]
)
```
rather than:
```elixir
foo(
  a ++
    b = [
  c,
  d
]
)
```

But when it's after an operator, suddenly this indentation on the initial line is not considered

**Comment by josevalim** [comment]
@michalmuskala sorry, just to be clear, the following is the expected behaviour in this case, right?

```
foo(
  a ++
    b = [
    c,
    d
  ]
)
```

Because it behaves the same as:

```
a ++
  b = [
  c,
  d
]
```

**Comment by josevalim** [comment]
Closing per the above, let me know if I got it wrong. :)
